libvast/src/system/index.cpp

```diff
@@ -976,11 +961,9 @@ index(index_actor::stateful_pointer<index_state> self,
             if (candidates.empty()) {
               VAST_DEBUG("{} returns without result: no partitions qualify",
                          *self);
-              no_result();
-              // TODO: When updating to CAF 0.18, remove the use of the
-              // untyped response promise and call deliver without arguments.
-              auto& untyped_rp = static_cast<caf::response_promise&>(rp);
-              untyped_rp.deliver(caf::unit);
+              self->send(self, atom::worker_v, worker);
+              rp.deliver(query_id, 0u, 0u);
+              caf::anon_send(client, atom::done_v);
```

Comment:
Why hide the index as the sender?

Reply:
I just needed a way to send from a typed actor to `client`. We could also make `client` a typed `receiver_actor<atom::done>`. Another approach would be to skip the `done` if no partitions qualify, but that is a protocol change again and the `query_processor` state machine would have to be adapted.